NIFI-8377 Fix CSVReader quoting and trimming with value separator inconsistency

Kasonnara opened the pull request:
#### Description of PR

fixes [NIFI-8377](https://issues.apache.org/jira/browse/NIFI-8377)

Add _withIgnoreSurroundingSpaces_ setting to CSVReader's format when trimming is enabled.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [X ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [X ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [X ] Have you verified that the full build is successful on JDK 11?
- [ X] ~~If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?~~ 
- [X] ~~If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?~~
- [X] ~~If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?~~
- [X] ~~If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?~~

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

Woutifier commented:
Hi Kasonnara. The tests don't pass.

This PR would change the default behavior of the CSV reader, that sounds (too) risky to me. Especially as trimming and processing double quotes if the field does not start with quotes seems at odds with RFC4180 ([](https://tools.ietf.org/html/rfc4180)).

The best course of action if you require this behavior is to introduce a new property to enable/disable this setting, and keeping the default at false (i.e. the behavior of the CSV reader will not change if nifi users make no changes except for updating nifi).

Kasonnara commented:
I apologize for the late reply, the last months were quite busy and then it took me some time to find out why I didn't replicate the CI test results on my window (some issues with long path, env vars, deps download that fail or test that failed because the language of my computed isn't english). Well in the end I switched to a another computer with Linux that I control more and everything went better.

As requested I added a property (disabled by default) to control the use of `withIgnoreSurroundingSpaces` and I had to rebase the commits or else some deps wouldn't be found anymore.

Kasonnara commented:
Oh my bad, for the last build error, I just dumb. I fix it immediately.
